imap_processing/cdf/tests/test_cdf_attribute_manager.py

```diff
@@ -1,58 +1,322 @@
+    cdf_manager.load_global_attributes("imap_test_global.yaml")
+
+    # Loading in instrument specific attributes
+    test_get_global_attrs = cdf_manager.get_global_attributes("imap_test_T1_test")
+
+    # Testing information previously loaded into global attributes (first if statement)
+    assert test_get_global_attrs["Project"] == "STP>Solar-Terrestrial Physics"
     assert (
-        mag_l1a_global_attrs["Mission_group"]
+        test_get_global_attrs["Source_name"]
         == "IMAP>Interstellar Mapping and Acceleration Probe"
     )
-    assert mag_l1a_global_attrs["Descriptor"] == "MAG>Magnetometer"
-    assert mag_l1a_global_attrs["Logical_source"] == "imap_mag_l1a_norm-raw"
+    assert test_get_global_attrs["Mission_group"] == "Dysfunctional Cats"
+    # Testing instrument specific global attributes (first elif statement)
+    assert test_get_global_attrs["Descriptor"] == "TEST>Testinstrument"
+    assert test_get_global_attrs["Data_type"] == "T1_test-one>Test-1 test one"
+    assert test_get_global_attrs["Logical_source"] == "imap_test_T1_test"
+    assert (
+        test_get_global_attrs["Logical_source_description"]
+        == "IMAP Mission TEST one document Level-T1."
+    )
+    # Not given, and not required information
+    assert test_get_global_attrs["Data_level"] is None
+    with pytest.raises(KeyError):
+        assert test_get_global_attrs["bad_name"] == "false info"
+
+    # Testing second elif statement
+    test_error_elif = cdf_manager.get_global_attributes("imap_test_T3_test")
+    with pytest.raises(KeyError):
+        assert test_error_elif["Data_type"] == "Does Not Exist"
+
+    # Load in more data using get_global_attributes
+    test_get_global_attrs_2 = cdf_manager.get_global_attributes("imap_test_T2_test")
+    # Testing information previously loaded into global attributes (first if statement)
+    assert test_get_global_attrs_2["Project"] == "STP>Solar-Terrestrial Physics"
+    # Testing first elif statement
+    assert test_get_global_attrs_2["Descriptor"] == "TEST>Testinstrument"
+    # "Data_type" not required according to default schema
+    assert test_get_global_attrs_2["Data_type"] == "T2_test-two>Test-2 test two"
+
+    # Testing how instrument_id operates
+    assert test_get_global_attrs["Project"] == cdf_manager.global_attributes["Project"]
+    assert (
+        test_get_global_attrs["Source_name"]
+        == cdf_manager.global_attributes["Source_name"]
+    )
+    assert (
+        test_get_global_attrs["Data_type"]
+        == cdf_manager.global_attributes["imap_test_T1_test"]["Data_type"]
+    )
+    assert (
+        cdf_manager.global_attributes["imap_test_T1_test"]["Logical_source"]
+        == "imap_test_T1_test"
+    )
+    with pytest.raises(KeyError):
+        assert cdf_manager.global_attributes["imap_test_T1_test"]["Project"]
+
+    # Trying to update a default global using get_global_attributes does not work.
```

Comment:
I don't really understand this comment... if its a note to self, it should be removed, or if it's a genuine note, could you expand on what you mean?